pkg/types/builder.go

```diff
@@ -76,12 +74,12 @@ func (g *Builder) buildResource(res *schema.Resource, names ...string) (*types.N
 		switch {
 		case sch.Computed && !sch.Optional:
 			obsFields = append(obsFields, field)
-			obsTags = append(obsTags, fmt.Sprintf("json:\"%s\" tf:\"%s\"", lowerCamelFieldName, snakeFieldName))
+			obsTags = append(obsTags, fmt.Sprintf("json:\"%s\" tf:\"%s\"", fieldName.LowerCamel, fieldName.Snake))
 		default:
 			if sch.Optional {
-				paramTags = append(paramTags, fmt.Sprintf("json:\"%s,omitempty\" tf:\"%s\"", lowerCamelFieldName, snakeFieldName))
+				paramTags = append(paramTags, fmt.Sprintf("json:\"%s,omitempty\" tf:\"%s\"", fieldName.LowerCamel, fieldName.Snake))
```

Comment:
I don't really have a strong opinion about it but it might have some side effects that are not related to this PR. So I'd prefer changing that in a separate PR, possibly in your late init PR where you can experiment with it?